sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs

```diff
@@ -562,6 +603,155 @@ public virtual DirectoryClient GetSubdirectoryClient(string subdirectoryName)
+        /// Optional SMB properties to set for the directory.
+        /// </param>
+        /// <param name="filePermission">
+        /// Optional file permission to set for the directory.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{StorageFileInfo}"/> describing the
+        /// state of the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual async Task<Response<StorageDirectoryInfo>> SetHttpHeadersAsync(
+            FileSmbProperties? smbProperties = default,
+            string filePermission = default,
+            CancellationToken cancellationToken = default) =>
+            await this.SetHttpHeadersInternal(
+                smbProperties,
+                filePermission,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="SetHttpHeadersInternal"/> operation sets system
+        /// properties on the directory.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/set-directory-properties"/>.
+        /// </summary>
+        /// <param name="smbProperties">
+        /// Optional SMB properties to set for the directory.
+        /// </param>
+        /// <param name="filePermission">
+        /// Optional file permission to set ofr the directory.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{StorageFileInfo}"/> describing the
+        /// state of the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<StorageDirectoryInfo>> SetHttpHeadersInternal(
+            FileSmbProperties? smbProperties,
+            string filePermission,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (this.Pipeline.BeginLoggingScope(nameof(DirectoryClient)))
+            {
+                this.Pipeline.LogMethodEnter(
+                    nameof(DirectoryClient),
+                    message:
+                    $"{nameof(this.Uri)}: {this.Uri}\n" +
+                    $"{nameof(smbProperties)}: {smbProperties}\n" +
```

Comment:
This doesn't have a meaningful `ToString`

Reply:
fixed